src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java

```diff
@@ -0,0 +1,287 @@
+package com.dotmarketing.business.cache.provider.caffine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.dotcms.enterprise.cache.provider.CacheProviderAPI;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.cache.provider.CacheProvider;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+
+public class CaffineCache extends CacheProvider {
+
+    private static final long serialVersionUID = 1348649382678659786L;
+
+    private Boolean isInitialized = false;
+
+    static final String DEFAULT_CACHE = CacheProviderAPI.DEFAULT_CACHE;
+    static final String LIVE_CACHE_PREFIX = CacheProviderAPI.LIVE_CACHE_PREFIX;
+    static final String WORKING_CACHE_PREFIX = CacheProviderAPI.WORKING_CACHE_PREFIX;
+
+    private final ConcurrentHashMap<String, Cache<String, Object>> groups =
+            new ConcurrentHashMap<>();
+    private final HashSet<String> availableCaches = new HashSet<>();
+
+
+    @Override
+    public String getName() {
+        return "Caffine Memory Cache";
+    }
+
+    @Override
+    public String getKey() {
+        return "LocalCaffineMem";
+    }
+
+    @Override
+    public void init() {
```

Comment:
There is at least two more CacheProviders (TimeCachedProvider, GuavaCache) with a very similar code, if not the same, for the init. Thoughts @freddyucv , @jdotcms, @agomez-dotcms, @jgambarios